src/Synth/OscilGen.cpp

```diff
@@ -1059,18 +1059,21 @@ short int OscilGen::get(float *smps, float freqHz, int resonance)
     clearAll(outoscilFFTfreqs, synth.oscilsize);
 
     int nyquist = (int)(0.5f * synth.samplerate_f / fabsf(freqHz)) + 2;
+    int nyquistLimited = (int)(0.5f * synth.samplerate_f / fabsf(freqHz) * nyquistFactor) + 2;
     if(ADvsPAD)
         nyquist = (int)(synth.oscilsize / 2);
     if(nyquist > synth.oscilsize / 2)
         nyquist = synth.oscilsize / 2;
+    if(nyquistLimited > synth.oscilsize / 2)
+        nyquistLimited = synth.oscilsize / 2;
 
     //Process harmonics
     {
         int realnyquist = nyquist;
 
         if(Padaptiveharmonics != 0)
             nyquist = synth.oscilsize / 2;
-        for(int i = 1; i < nyquist - 1; ++i)
+        for(int i = 1; i < nyquistLimited - 1; ++i)
```

Comment:
`Padaptiveharmonics` has no effect anymore now.

Reply:
> `Padaptiveharmonics` has no effect anymore now.

it is also used inside OscilGen::adaptiveharmonic()
look at line 913

Reply:
What I really meant was that the 2 lines above this line do not have an effect on this for loop anymore. Which means that this code breaks the current behavior.